### PR TITLE
Add allow_multiple_attempts setting to Match and FreeResponse levels

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds a field called `allow_multiple_attempts` to `FreeResponse` and `Match` levels. `Multi` is a subclass of `Match` so it will also inherit this field.

I updated the `FreeResponse` editor to allow for editing this field. I tested this manually, though I don't know a way of adding tests for this file.
Screen Shot 2023-03-03 at 10 47 23 AM

Updating the editor for `Match` levels was tougher. Because `Match` is a `DSLDefined` level (also I thought until this week that `FreeResponse` levels were also `DSLDefined` woops), the editor is not user friendly. I could not figure out a way to make a friendly input for this field in the editor, so I added it to the default DSL. For existing levels, editors will need to add `allow_multiple_attempts true` to the DSL editor.

Screen Shot 2023-03-03 at 10 49 29 AM

This field isn't being used yet -- it's just being added early so that curriculum writers can set this field now.

### CODE REVIEW

**Review by dju90 — COMMENTED** [review]
Yikes about the DSL level UI, but that notwithstanding, looks good to me!

File: dashboard/app/models/levels/match.rb [hunk 1]
```diff
@@ -25,13 +25,18 @@
...
+  )
+
   def dsl_default
     <<~RUBY
       name '#{DEFAULT_LEVEL_NAME}'
       title 'title'
       description 'description here'
       question 'Question'
       answer 'Answer 1'
+      allow_multiple_attempts nil
```
Comment (hannahbergam) [inline dashboard/app/models/levels/match.rb:39]: 🫠

**Review by hannahbergam — APPROVED** [review]
🤷🏻‍♀️ I'm not all that familiar with this code so am not sure how else you would've added that field for dsl levels! If it works then I'm game

**Comment by dju90** [comment]
Non-blocking question: are `DSLDefined` Match levels in haml for the user-facing view?

**Comment by bethanyaconnor** [comment]
> are DSLDefined Match levels in haml for the user-facing view?

Yes. Though, interestingly, they can't be used as contained levels

**Pull request merged**